src/include/function/scalar_function.h

```diff
@@ -53,6 +53,16 @@ struct ScalarFunction final : public BaseScalarFunction {
           execFunc{std::move(execFunc)}, selectFunc(std::move(selectFunc)),
           compileFunc{std::move(compileFunc)} {}
 
+    ScalarFunction(std::string name, std::vector<common::LogicalTypeID> parameterTypeIDs,
```

Comment:
I think we are having way too many constructors here. Maybe we can just do
```
auto f = ScalarFunction(name, paramTypes, returnType)
f.bindFunc = bindFunc
```

Reply:
I think it is fine because i saw duckdb also has set of constructors.